src/lava/magma/core/model/py/model.py

```diff
@@ -237,3 +128,192 @@ def _handle_set_var(self):
+            return
+        if self.lrn_guard() and self.pre_guard():
+            self.process_to_service.send(
+                PyLoihiProcessModel.Response.REQ_PRE_LRN_MGMT)
+        elif self.lrn_guard():
+            self.process_to_service.send(
+                PyLoihiProcessModel.Response.REQ_LEARNING)
+        elif self.post_guard():
+            self.process_to_service.send(
+                PyLoihiProcessModel.Response.REQ_POST_LRN_MGMT)
+        else:
+            self.process_to_service.send(PyLoihiProcessModel.Response.STATUS_DONE)
+
+    def _pre_mgmt(self):
+        self.phase = PyLoihiProcessModel.Phase.PRE_MGMT
+        if self.pre_guard():
+            self.run_pre_mgmt()
+        if self._req_pause or self._req_stop:
+            self._handle_pause_or_stop()
+            return
+        self.process_to_service.send(PyLoihiProcessModel.Response.STATUS_DONE)
+
+    def _post_mgmt(self):
+        self.phase = PyLoihiProcessModel.Phase.POST_MGMT
+        if self.post_guard():
+            self.run_post_mgmt()
+        if self._req_pause or self._req_stop:
+            self._handle_pause_or_stop()
+            return
+        self.process_to_service.send(PyLoihiProcessModel.Response.STATUS_DONE)
+
+    def _lrn(self):
+        self.phase = PyLoihiProcessModel.Phase.LRN
+        if self.lrn_guard():
+            self.run_lrn()
+        if self._req_pause or self._req_stop:
+            self._handle_pause_or_stop()
+            return
+        if self.post_guard():
+            self.process_to_service.send(
+            PyLoihiProcessModel.Response.REQ_POST_LRN_MGMT)
+            return
+        self.process_to_service.send(PyLoihiProcessModel.Response.STATUS_DONE)
+
+    def _host(self):
+        self.phase = PyLoihiProcessModel.Phase.HOST
+
+    def _stop(self):
+        self.process_to_service.send(PyLoihiProcessModel.Response.STATUS_TERMINATED)
+        self.join()
+
+    def _pause(self):
+        self.process_to_service.send(PyLoihiProcessModel.Response.STATUS_PAUSED)
+
+    def _handle_pause_or_stop(self):
+        if self._req_pause:
+            self._req_rs_pause()
+        elif self._req_stop:
+            self._req_rs_stop()
+
+    def _req_rs_pause(self):
+        self._req_pause = False
+        self.process_to_service.send(PyLoihiProcessModel.Response.REQ_PAUSE)
+
+    def _req_rs_stop(self):
+        self._req_stop = False
+        self.process_to_service.send(PyLoihiProcessModel.Response.REQ_STOP)
+
+    def run(self):
+        """Retrieves commands from the runtime service to iterate through the
+        phases of Loihi and calls their corresponding methods of the
+        ProcessModels. The phase is retrieved from runtime service
+        (service_to_process). After calling the method of a phase of all
+        ProcessModels the runtime service is informed about completion. The
+        loop ends when the STOP command is received."""
+        if enum_equal(self.phase, PyLoihiProcessModel.Phase.PRE_MGMT) or \
```

Comment:
Add a comment on what we are doing here since it is something special about Ref-/VarPorts